### PR TITLE
EXAPP-103-Marketing-List

### PR DESCRIPTION
**Pull request by vyomsoni**
I created a Marketing List component in this branch.

I made the 4 LWC components:-

1) -> marketingListCmp
2) -> marketingListFilterAddCmp
3) -> marketingListFilterCmp
4) -> markeitngListTileView

2 Apex classes :-

1) -> MarketingListCmpController
2) -> MarketingListFilterController

1 Custom metadata :-

1) -> MarketingListFilterConfig

1 fieldset on contact object :-

1) -> MarketingListFieldSet

### CODE REVIEW

File: force-app/main/default/classes/MarketingListFilterController.cls [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+        if (!Schema.MVEX__MarketingListFilterConfig__mdt.SObjectType.getDescribe().isAccessible()) {
+            throw new AuraHandledException('You do not have permission to access MarketingListFilterConfig__mdt.');
+        }
+        // Check if the user has read access to StaticFields__c field
+        if (!Schema.MVEX__MarketingListFilterConfig__mdt.MVEX__StaticFields__c.getDescribe().isAccessible()) {
+            throw new AuraHandledException('You do not have permission to access StaticFields__c.');
+        }
+        
+        // Perform the SOQL query
+        MVEX__MarketingListFilterConfig__mdt filterConfig = [SELECT MVEX__StaticFields__c FROM MVEX__MarketingListFilterConfig__mdt LIMIT 1];
```
Comment (MRG-MV) [inline force-app/main/default/classes/MarketingListFilterController.cls:336]: CRUD permission before SOWL/DML missing.

File: force-app/main/default/lwc/marketingListCmp/marketingListCmp.js [hunk 2]
```diff
@@ -0,0 +1,521 @@
...
+    * Created By:Vyom Soni
+    */
+    goToPrevFeaturedProperty() {
+        if (this.pageNumber > 1) {
+            this.pageNumber--;
+            this.updateProcessedContactData();
+            this.updatePaginationButtons();
+            setTimeout(() => {
+                this.scrollToTop();
+            }, 0);
```
Comment (MRG-MV) [inline force-app/main/default/lwc/marketingListCmp/marketingListCmp.js:487]: Change this don't use settimeout ask Karan Singh for help on this if you don't get a solution

File: force-app/main/default/lwc/marketingListCmp/marketingListCmp.js [hunk 3]
```diff
@@ -0,0 +1,521 @@
...
+    * Created By:Vyom Soni
+    */
+    goToNextFeaturedProperty() {
+        if (this.pageNumber < this.totalPages) {
+            this.pageNumber++;
+            this.updateProcessedContactData();
+            this.updatePaginationButtons();
+            setTimeout(() => {
+                this.scrollToTop();
+            }, 0);
```
Comment (MRG-MV) [inline force-app/main/default/lwc/marketingListCmp/marketingListCmp.js:504]: Change this don't use settimeout ask Karan Singh for help on this if you don't get a solution

File: force-app/main/default/lwc/marketingListTileView/marketingListTileView.js [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+    }
+ 
+    /**
+     * Method Name : set contactsdata
+     * @description : set the filtered contact data from contact manager component reactively
+     * @param: value- data from the parent component
+     * date:24/06/2024
+     * Created By: Vyom Soni
+    */
+    set contactsdata(value) {
```
Comment (MRG-MV) [inline force-app/main/default/lwc/marketingListTileView/marketingListTileView.js:30]: use @api decorater and api methods for this to get updated data from the parent

File: force-app/main/default/lwc/marketingListTileView/marketingListTileView.js [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+    * Created By: Vyom Soni
+    * date:24/06/2024
+    */
+    goToPrevFeaturedProperty() {
+        if (this.pageNumber > 1) {
+            this.pageNumber--;
+            this.updateProcessedContactData();
+            this.updatePaginationButtons();
+        }
+        setTimeout(() => {
```
Comment (MRG-MV) [inline force-app/main/default/lwc/marketingListTileView/marketingListTileView.js:178]: remove set time out

File: force-app/main/default/lwc/marketingListTileView/marketingListTileView.js [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+    * @description : update the page number when the next button is clicked
+    * Created By: Vyom Soni
+    * date:24/06/2024
+    */
+    goToNextFeaturedProperty() {
+        if (this.pageNumber < this.totalPages) {
+            this.pageNumber++;
+            this.updateProcessedContactData();
+            this.updatePaginationButtons();
+            setTimeout(() => {
```
Comment (MRG-MV) [inline force-app/main/default/lwc/marketingListTileView/marketingListTileView.js:194]: remove set timeout

**Review by MRG-MV — CHANGES_REQUESTED** [review]
Make requested changes

**Review by MRG6804 — APPROVED** [review]

**Pull request merged**